ibis/expr/api.py

```diff
@@ -389,23 +392,53 @@ def timestamp(
     TimestampScalar
         A timestamp expression
     """
-    if isinstance(value, str):
-        try:
-            value = pd.Timestamp(value, tz=timezone)
-        except pd.errors.OutOfBoundsDatetime:
-            value = dateutil.parser.parse(value)
-    if isinstance(value, numbers.Integral):
-        raise TypeError(
-            (
-                "Passing an integer to ibis.timestamp is not supported. Use "
-                "ibis.literal({value}).to_timestamp() to create a timestamp "
-                "expression from an integer."
-            ).format(value=value)
-        )
+    raise NotImplementedError(f'cannot convert {type(value)} to timestamp')
+
+
+@timestamp.register(np.int8)
+@timestamp.register(np.int16)
+@timestamp.register(np.int32)
+@timestamp.register(np.int64)
+@timestamp.register(np.uint8)
+@timestamp.register(np.uint16)
+@timestamp.register(np.uint32)
+@timestamp.register(np.uint64)
+@timestamp.register(np.float32)
+@timestamp.register(np.float64)
```

Comment:
```suggestion
@timestamp.register(np.integer)
@timestamp.register(np.floating)
```

Reply:
I looked for this and couldn't find it!  Thanks.